src/server.js

```diff
@@ -0,0 +1,150 @@
+"use strict";
+var http = require('http');
+var graphql = require('graphql');
+require('babel-register');
+var WebSocketServer = require('websocket').server;
+var graphql_tools = require('graphql-tools');
+var graphql_validator = require('graphql/validation');
+var graphql_execution = require('graphql/execution');
+
+class Server {
+  /*
+  options {
+    schema: GraphQLSchema
+    contextValue?: any,
+    rootValue?: any,
+    formatResponse?: (Object) => Object,
+    validationRules?: Array<any> 
+    triggerGenerator?: (Object) => [Object]
+  }
+  */
+  constructor(options, httpServer) {
+    this.options = options;
+    this.triggers = {}; // trigger_object: [{connection, sub_id}]
+    //initialize http server
+
+    //init and connect websocket server to http
+    this.wsServer = new WebSocketServer({
+      httpServer: httpServer,
+      autoAcceptConnections: false
+    });
+
+    this.wsServer.on('request', (request) => {
+      //accept connection
+      var connection = request.accept('graphql-protocol', request.origin);
+      console.log('Accepted connection');
+      connection.on('message', (message) => {
+        let message_data = JSON.parse(message.utf8Data);
+        if (message_data.type === 'subscription_start') {
+          let syntax_errors = graphql_validator.validate(this.options.schema, graphql.parse(message_data.query), this.options.validationRules);
+          if (syntax_errors.length > 0) {
+            let message = {
+              type: 'subscription_fail',
+              errors: syntax_errors,
+              id: message_data.id,
+            };
+            connection.sendUTF(JSON.stringify(message));
+          } else {
+            const sub_id = message_data.id;
+            if (!connection.subscriptions) {
+              connection.subscriptions = {};
```

Comment:
I'm now just storing all necessary information in server.triggers (inc the connection object), and not storing anything on the connection